### PR TITLE
feat(FX-3707): show a warning message for a duplicate alert

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3707]

### Acceptance Criteria
* If the user clicks on the "Cancel" button, then
  * the warning message should close
  * the duplicate alert will not be replaced
  * the new alert will not be created (for create flow), new changes should not be applied to the old alert (for update flow)
* If the user clicks on the "Replace" button, then
  * the warning message should close
  * the duplicate alert should be replaced with a new alert.
* Warning message
  * Title: Duplicate Alert
  * Message: You already have a Saved Alert with these filters. Do you want to replace it?
  * Buttons
    * Cancel
    * Replace

#### iOS - Create flow
https://user-images.githubusercontent.com/3513494/151366055-0a5c0800-225e-44f0-8dc4-e3b1f97b6829.mp4

#### iOS - Update flow
https://user-images.githubusercontent.com/3513494/151366086-eb6fdda8-8a42-49d4-9ae1-05e655d91665.mp4

#### Android - Create flow
https://user-images.githubusercontent.com/3513494/151366172-4b3e28f3-3990-4c94-8b3f-5315e3353ade.mp4

#### Android - Update flow
https://user-images.githubusercontent.com/3513494/151366207-b602e49b-3803-469c-935a-84e2798bb5b8.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Show a warning message for a duplicate warning - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3707]: https://artsyproduct.atlassian.net/browse/FX-3707?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Show a warning message for a duplicate warning - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 891ca7b89e1fc90357b4fe57a1b1eafff3a7d27e
</p>

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export const SavedSearchAlertForm: React.FC<SavedSearchAlertFormProps> = (props)
           return
         }
 
-        submitHandler(userAlertSettings, clearedAttributes)
+        await submitHandler(userAlertSettings, clearedAttributes)
```
Comment (gkartalis) [inline src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:127]: 💯

**Review by gkartalis — APPROVED** [review]
Looks great to me! 🚀  great test cases as well 🥇

**Review by gkartalis — APPROVED** [review]

**Review by mdole — APPROVED** [review]
nice! two minor suggestions, otherwise looking good

File: src/lib/Scenes/SavedSearchAlert/Components/Form.tsx [hunk 2]
```diff
@@ -62,7 +60,7 @@ export const Form: React.FC<FormProps> = (props) => {
   let isSaveAlertButtonDisabled = false
 
   // Data has not changed or has already been saved
-  if ((isEditMode && !dirty) || isPreviouslySaved) {
+  if (isEditMode && !dirty) {
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/Components/Form.tsx:63]: Good point!

File: src/lib/Scenes/SavedSearchAlert/helpers.ts [hunk 3]
```diff
@@ -107,3 +107,25 @@ export const clearDefaultAttributes = (attributes: SearchCriteriaAttributes) =>
...
 }
+
+export const showWarningMessageForDuplicateAlert = ({
+  onReplacePress,
+}: {
+  onReplacePress: () => void
+}) => {
+  Alert.alert(
+    "Duplicate Alert",
+    "You already have a Saved Alert with these filters. Do you want to replace it?",
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/helpers.ts:118]: Clarified this moment w/ Lois [here](https://artsy.slack.com/archives/C9SATFLUU/p1643295785390300?thread_ts=1643291203.384700&cid=C9SATFLUU).

**Pull request merged**